source/module_io/CMakeLists.txt

```diff
@@ -96,10 +96,14 @@ if(ENABLE_COVERAGE)
 endif()
 
 if(BUILD_TESTING)
-  add_subdirectory(test)
-  add_subdirectory(test_serial)
+  if(ENABLE_MPI)
+    add_subdirectory(test)
+    add_subdirectory(test_serial)
```

Comment:
why test_serial cannot build without MPI?

Reply:
It just can't get built
![image](https://github.com/deepmodeling/abacus-develop/assets/23211911/db624e46-fd7c-4fbf-a81e-48ee3ead141c)
I dont know why and I'm not familiar this part. So I just removed this as a temporary solution.
It may because for quite a long past time, disabling MPI will also disable BUILD_TESTING at the same time. So no one maintained this part of code since then.